fqlx/queries/queryBuilder.js

```diff
@@ -0,0 +1,18 @@
+const { fql } = require('fauna')
+const { createUpdateFunction }  = require('./function')
+
+module.exports = ({
+  functions = [],
+}) => {
+  const queries = [
+    ...functions.map(f => createUpdateFunction(f)),
+  ]
+
+  const reduced = queries
+    .filter((q) => !!q)
+    .reduce((res, curr) => {
+      return fql`${res}\n\n${curr}`
+    }, fql`let results = []`)
```

Comment:
I think you can avoid appending each time, if each `createUpdateFunction` returns one result:
```js
const queries = [
  ...functions.map(f => createUpdateFunction(f))
];

return fql`${queries}`
```

then `createUpdateFunction` would just return an object like this:
```js
  fql`
  if (Function.byName(${params.name}) != null) {
    let f = Function.byName(${params.name})
    if (${and("f", params)}) {
      { type: "function", name: ${params.name}, result: "not changed" }
    } else {
      f.update(${params})
      { type: "function", name: ${params.name}, result: "updated" }
    }
  } else {
    Function.create(${params})
    { type: "function", name: ${params.name}, result: "created" }
  }`
```

if any of the queries need to return multiple results, then feel free to leave it as-is.

Reply:
Each should ideally do one thing, so this seems like a nice improvement